src/main/java/hudson/plugins/performance/data/ReportValueSelector.java

```diff
@@ -0,0 +1,74 @@
+package hudson.plugins.performance.data;
+
+import hudson.model.AbstractProject;
+import hudson.model.Job;
+import hudson.plugins.performance.PerformancePublisher;
+import hudson.plugins.performance.reports.AbstractReport;
+
+public abstract class ReportValueSelector {
+
+  public abstract long getValue(AbstractReport report);
+  public abstract String getGraphType();
+
+  public static ReportValueSelector get(Job<?, ?> job) {
```

Comment:
please, make code style of your classes as well as in the whole project (use 4 spaces instead of 2)